pkg/scheduler/dependency_graph.go

```diff
@@ -434,37 +482,74 @@ func (sched *Scheduler) BuildDependencyGraph(options interfaces.DependencyGraphO
 	depGraph := NewDependencyGraph(sched, options)
 	rootContext := &GraphContext{scheduler: sched, graph: depGraph, flow: flow, args: options.Args}
 
-	if _, ok := dependencies[fullFlowName]; !ok {
-		log.Printf("Flow %s is empty", options.FlowName)
-		return depGraph, nil
+	replicas, deleteReplicas, err := sched.allocateReplicas(flow, options)
+	if err != nil {
+		return nil, err
 	}
 
-	type Block struct {
-		dependency        client.Dependency
-		scheduledResource *ScheduledResource
-		parentContext     *GraphContext
+	err = sched.fillDependencyGraph(rootContext, resDefs, dependencies, flow, replicas, false)
+	if err != nil {
+		return nil, err
 	}
-
-	replicas, err := sched.allocateReplicas(flow, options)
+	err = sched.fillDependencyGraph(rootContext, resDefs, dependencies, flow, deleteReplicas, true)
 	if err != nil {
 		return nil, err
 	}
 
+	for _, value := range depGraph.graph {
+		value.RequiredBy = unique(value.RequiredBy)
+		value.Requires = unique(value.Requires)
+		value.usedInReplicas = unique(value.usedInReplicas)
+	}
+
+	if len(deleteReplicas) > 0 {
```

Comment:
because if we don't have deleteReplicas then we have nothing to delete. The code in the block creates additional graph that consists of all existing replicas (both those that we want to delete and those that will remain) so that we could find out which resources belong to the deleted replicas and also exclude resources that shared between all replicas and there are still alive replicas that have them. Together with destruction graph (graph of resources that are created *for the destruction*) this gives enough information to build list of resources that need to be deleted. If len(deleteReplicas) == 0, then there is nothing to delete so nothing of above need to be done. I'm not sure that all of this can be explained with a function name. I'm don't even know how to call such function